cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+    }
+
+    private String getTableLocation(String sourceDb,String hiveConnection,String sourceStgTable) throws Exception{
+        Connection conn = getHiveJDBCConnection(sourceDb,hiveConnection);
+        Statement st = conn.createStatement();
+        ResultSet rs = st.executeQuery("desc formatted "+sourceStgTable);
+        String tableLocation=new String();
+        while (rs.next()) {
+            if(rs.getString(1).trim().equals("Location:")) tableLocation=rs.getString(2);
+        }
+        return tableLocation;
+    }
+
+    private MigrationPreprocessorInfo setOozieUtilProperties(String sourceDb,String destDb,String sourceStgtable,String table, String sourceJobTrackerAddress, String sourceNameNodeAddress, String destNameNodeAddress, String srcHiveConnection,String processId, String instanceExecId, MigrationPreprocessorInfo migrationPreprocessorInfo){
+        String stgPartitionsWithoutDataTypes=removeDataTypesFromColumnList(stgPartitionColumns);
+        String stgRegColumnsWithoutDataTypes=removeDataTypesFromColumnList(sourceRegularColumns);
+        migrationPreprocessorInfo.setSrcStgDb(sourceDb);
+        migrationPreprocessorInfo.setSrcStgTable(sourceStgtable);
+        migrationPreprocessorInfo.setStgAllPartCols(stgPartitionsWithoutDataTypes);
+        migrationPreprocessorInfo.setSrcRegularCols(stgRegColumnsWithoutDataTypes);
+        migrationPreprocessorInfo.setSrcBPCols(stgPartitionsWithoutDataTypes.substring(0,stgPartitionsWithoutDataTypes.lastIndexOf(",")));
+        migrationPreprocessorInfo.setSrcDb(sourceDb);
+        migrationPreprocessorInfo.setSrcTable(table);
+        migrationPreprocessorInfo.setFilterCondition(filterCondition);
+        migrationPreprocessorInfo.setJtAddress(sourceJobTrackerAddress);
+        migrationPreprocessorInfo.setNnAddress(sourceNameNodeAddress);
+        //replacing the source namenode(quickstart.cloudera or sandbox.hortonworks.com) with actual public ip obtained from through properties (src-nn)
+        migrationPreprocessorInfo.setSrcStgTablePath(sourceNameNodeAddress+srcStgTableLocation.substring(srcStgTableLocation.indexOf("/",7),srcStgTableLocation.length())+"/");
+        migrationPreprocessorInfo.setDestStgFolderPath(destNameNodeAddress+"/tmp/"+processId+"/"+instanceExecId);
+        migrationPreprocessorInfo.setDestStgFolderContentPath(destNameNodeAddress+"/tmp/"+processId+"/"+instanceExecId+"/");
+        migrationPreprocessorInfo.setDestTablePath(destTableLocation);
+        migrationPreprocessorInfo.setDestDb(destDb);
+        migrationPreprocessorInfo.setDestTable(table);
+        migrationPreprocessorInfo.setDestFileSystem(destNameNodeAddress);
+        return migrationPreprocessorInfo;
+    }
+
+    public Properties getParams(String pid, String configGroup) {
+        GetProperties getProperties = new GetProperties();
+        java.util.Properties listForParams = getProperties.getProperties(pid, configGroup);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "listForParams". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1488)